src/Runtime/ExecutionSession.hpp

```diff
@@ -29,6 +29,19 @@ using queryEntryPointsFuncType = const char **(*)(int64_t *);
 using signatureFuncType = const char *(*)(const char *);
 using OMTensorUniquePtr = std::unique_ptr<OMTensor, decltype(&omTensorDestroy)>;
 
+/* ExecutionSession
+ * Class that supports executing compiled models.
+ *
+ * When the execution session does not work for known reasons, this class will
+ * throw std::runtime_error errors. Errno info will provide further info about
+ * the specific error that was raised.
+ *
+ * EDOM when it could not load the library or a needed symbol was not found.
+ * EINVAL when it expected an entry point prior to executing a specific
+ * function.
+ * EPERM when the model executed on a machine without a compatible
+ * hardware/specialzed accelerator.
+ */
```

Comment:
I wonder if you would like to write a document about these errors and put it under the `docs` folder.

Reply:
Yes, I see that there is no doc whatsoever of the ExecutionSession, there are examples in readme for examples, but nothing solid.

Reply:
Will do, but let me do a new PR for a UsingC++Runtime.md where we can better show how to use the C++ interface.

Reply:
EDOM is a fairly uncommen errno to use. EFAULT is probably more common.

Reply:
changed to EFAULT as suggested